.github/workflows/conformance-ginkgo.yaml

```
@@ -39,16 +66,19 @@ concurrency:
   # - Event type
   # - A unique identifier depending on event type:
   #   - schedule: SHA
-  #   - workflow_dispatch: PR number
+  #   - workflow_dispatch: PR number. Because in workflow_call the parent context
+  #     is used, we need to add a prefix name ('ginkgo') for the concurrency group
+  #     so that workflows that are executed in parallel don't conflict with each
+  #     other.
   #
   # This structure ensures a unique concurrency group name is generated for each
   # type of testing, such that re-runs will cancel the previous run.
   group: |
     ${{ github.workflow }}
     ${{ github.event_name }}
     ${{
-      (github.event_name == 'schedule' && github.sha) ||
-      (github.event_name == 'workflow_dispatch' && github.event.inputs.PR-number)
+      (github.event_name == 'schedule' && format('{0}-{1}', github.sha, 'ginkgo')) ||
+      (github.event_name == 'workflow_dispatch' && format('{0}-{1}', github.event.inputs.PR-number, 'ginkgo'))
```

Incidentally another clearer option (if necessary) might be:

```suggestion
      (github.event_name == 'schedule' && github.sha) ||
      (github.event_name == 'workflow_dispatch' && github.event.inputs.PR-number) ||
      (github.event_name == 'workflow_call' && format('{0}-{1}', github.event.inputs.PR-number, 'race'))
```

---

This won't work due to https://github.com/cilium/cilium/pull/41541#discussion_r2378782808 .

However I still don't understand why the code change on these lines 80-81 needs to be made. The earlier comment says:

> ... we need to add a prefix name ('ginkgo') for the concurrency group so that workflows that are executed in parallel don't conflict with each other.

Adding a static string into an existing field will just make all callers of the workflow add that string to the jobs so I don't see how that prevents the conflict.

---

@joestringer let me know if the example is better. Also the reason why I added this suffix was because of the failure I saw with conformance-ipsec.yaml as you can see in [here](https://github.com/cilium/cilium/actions/runs/16441318660)

---

Oh, so this prevents conflict with ipsec but it doesn't prevent conflict between the ginkgo and ginkgo-race runs?

Seems like we should pull the workflow name into this group configuration for all workflows, maybe as a separate PR (+ backport?)

---

Huh except `${{ github.workflow }}` is already in the group configuration :confused:

---

> Oh, so this prevents conflict with ipsec but it doesn't prevent conflict between the ginkgo and ginkgo-race runs?
> 
> Seems like we should pull the workflow name into this group configuration for all workflows, maybe as a separate PR (+ backport?)

no no, the ipsec case was just an example where the ipsec.yaml executes conformance-aks/eks/gke and that was causing conflicts across the 3.

---

I'm confused. This PR is about extending ginkgo to be reusable and add ginkgo-race. I don't see how ipsec relates to this specific PR. If groups need to be fixed for all workflows we should handle that in a separate PR to make it super clear why and make all the workflows consistent.

---

Let's start over... :smile: 
I've added this comment.
```
  # The 'ginkgo' suffix added to both schedule and workflow_dispatch ensures
  # that for workflow_call events it doesn't conflict with other potential
  # parallel workflow_call done to other workflows from the same parent
  # workflow. As otherwise the group name would be identical.
```

The reason I mentioned the conformance-ipsec.yaml is to give you an example of what happens if we don't add the suffix. In conformance-ipsec.yaml, it workflow_calls aks, eks and gke yamls. As they didn't have a way to distinguish between themselves, the concurrency group name is the same for both 3 since they will have the same key: `Conformance IPsec (ci-ipsec) workflow_dispatch ec96c2b5ff9343fadb6fa9a3bb30e6b4b7b7707c`



> Huh except `${{ github.workflow }}` is already in the group configuration 😕

On an ideal world this would work but remember this is on a `workflow_call`. The `workflow_call` will not mention `Conformance Ginkgo (ci-ginkgo)`, it will mention the workflow that call this one. This is what happened with Conformance IPsec:  `Conformance IPsec (ci-ipsec) workflow_dispatch ec96c2b5ff9343fadb6fa9a3bb30e6b4b7b7707c`. This key was detected on the GKE workflow where you can see there's no mention of GKE as part of the key.

---

Oof. What you say now makes sense to me but it is counterintuitive. I made a couple of minor suggestions to help clarify this.

---

I will resolve this thread and leave it up to you on whether the alternative suggestions are easier to adopt and maintain or leave as is at your discretion.